tests/test_contrib_cern.py

```diff
@@ -194,3 +205,15 @@ def test_account_info_not_allowed_account(app, example_cern):
     assert resp.status_code == 302
     assert session['_flashes'][0][0] == 'danger'
     assert session['_flashes'][0][1] == 'CERN account not allowed.'
+
+
+def test_empty_disconnect_identity(app):
```

Comment:
@topless I am not sure I understand this test, what is its purpose?

Reply:
The initial problem was that when there was no value in AnonymousIdentity.provides, we were returning a dict instead of a set. Tests didn't catch it because there was always an entry in provides. The purpose of this test was to validate, that we pop the correct type from the session when there was nothing there. IMO is redundant especially being replicated every time for every provider and for both ui and rest.